src/AstrometryFit.cc

```diff
@@ -216,7 +216,7 @@ void AstrometryFit::LSDerivatives1(const CcdImage &ccdImage, TripletList &tList,
     /* this routine works in two different ways: either providing the
        ccdImage, of providing the MeasuredStarList. In the latter case, the
        ccdImage should match the one(s) in the list. */
-    if (msList) assert((*(msList->begin()))->ccdImage == &ccdImage);
+    if (msList) assert(&(msList->front()->getCcdImage()) == &ccdImage);
```

Comment:
`getCcdImage` dereferences the `_ccdImage` pointer held in the measuredStar. Also, `msList->begin()` is an iterator, while `msList->front()` is the first item, so we don't need the other deference there anymore.